core/mavlink_commands.h

```diff
@@ -30,20 +31,65 @@ class MAVLinkCommands
 
     typedef std::function<void(Result, float)> command_result_callback_t;
 
-    struct Params {
-        float v[7];
+    struct CmdInt {
+        uint8_t target_system_id;
+        uint8_t target_component_id;
+        MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+        uint16_t command;
+        bool current = 0;
+        bool autocontinue = false;
+        struct Params {
+            float param1 = NAN;
+            float param2 = NAN;
+            float param3 = NAN;
+            float param4 = NAN;
+            int32_t lat_deg = 0;
+            int32_t lon_deg = 0;
+            float alt_m = NAN;
```

Comment:
I just realized, we better name last 3 members as **x**, **y** and **z**., **x** could be in meters also in case of Local position. So, `_deg` would mislead. Right ?

x | int32_t | PARAM5 / local: x position in meters \* 1e4, global: latitude in degrees * 10^7
-- | -- | --
y | int32_t | PARAM6 / local: y position in meters * 1e4, global: longitude in degrees * 10^7
z | float | PARAM7 / z position: global: altitude in meters (relative or absolute, depending on frame.

Reply:
Ok, makes sense then